fedimint-client/src/lib.rs

```diff
@@ -1589,6 +1589,14 @@ impl ClientBuilder {
 
         let final_client = FinalClient::default();
 
+        // Re-derive client's root_secret using raw bytes from the provided root_secret
+        // and salt from the federation ID. This eliminates the possibility of having
+        // the same client root_secret across multiple federations.
+        let root_secret = DerivableSecret::new_root(
+            &root_secret.to_chacha20_poly1305_key_raw(),
+            &config.global.federation_id().0,
+        );
+
```

Comment:
1. Is this the correct approach—to get deterministic bytes from the provided `root_secret` and salt it with the byte representation of federation ID?
2. Is it fine to reuse `DerivableSecret::to_chacha20_poly1305_key_raw()` for this purpose? I also see `DerivableSecret::to_random_bytes()` but the "random" in the name made me a bit nervous about it being deterministic (even though it looks like it is?)

Reply:
`to_random_bytes` would be better imo, less confusing. The randomness there is deterministic, i.e. it's using a PRNG.

Reply:
Updated